contracts/TradetrustERC721.sol

```diff
@@ -13,60 +13,71 @@ contract TitleEscrowCreator is ITitleEscrowCreator {
   }
 }
 
-contract TradeTrustERC721 is TitleEscrowCreator, ERC721MintableFull, IERC721Receiver {
+interface ITradeTrustERC721 is IERC721Receiver {
+  // TODO: rename these to the appropriate names
+  function destroyToken(uint256 _tokenId) external;
+
+  function sendToNewTitleEscrow(
+    address beneficiary,
+    address holder,
+    uint256 _tokenId
+  ) external;
+
+  function sendToken(address to, uint256 _tokenId) external;
+}
+
+contract TradeTrustERC721 is TitleEscrowCreator, ERC721Mintable, IERC721Receiver {
   event TokenBurnt(uint256 indexed tokenId);
   event TokenReceived(address indexed operator, address indexed from, uint256 indexed tokenId, bytes data);
 
-  // ERC165: Interface for this contract, can be calculated by CalculateTradeTrustERC721Selector()
-  // Only append new interface id for backward compatibility
-  bytes4 private constant _INTERFACE_ID_TRADETRUST_ERC721 = 0x9f9e69f3;
-  
-  constructor(string memory name, string memory symbol) public ERC721MintableFull(name, symbol) {
-    // register the supported interface to conform to TradeTrustERC721 via ERC165
-    _registerInterface(_INTERFACE_ID_TRADETRUST_ERC721);
+  constructor(string memory name, string memory symbol) ERC721Mintable(name, symbol) { return; }
+
+  function supportsInterface(bytes4 interfaceId) public view virtual override(ERC721Mintable) returns (bool) {
+    return
+      interfaceId == type(ITitleEscrowCreator).interfaceId ||
+      interfaceId == type(ITradeTrustERC721).interfaceId ||
+      ERC721Mintable.supportsInterface(interfaceId);
   }
 
-  function onERC721Received(address _operator, address _from, uint256 _tokenId, bytes memory _data)
-    public
-    returns (bytes4)
-  {
+  function onERC721Received(
+    address _operator,
+    address _from,
+    uint256 _tokenId,
+    bytes memory _data
+  ) public override returns (bytes4) {
     emit TokenReceived(_operator, _from, _tokenId, _data);
     return this.onERC721Received.selector;
   }
 
   function destroyToken(uint256 _tokenId) public onlyMinter {
     require(ownerOf(_tokenId) == address(this), "Cannot destroy token: Token not owned by token registry");
     // Burning token to 0xdead instead to show a differentiate state as address(0) is used for unminted tokens
-    _safeTransferFrom(ownerOf(_tokenId), 0x000000000000000000000000000000000000dEaD, _tokenId, "");
     emit TokenBurnt(_tokenId);
+    this.safeTransferFrom(ownerOf(_tokenId), 0x000000000000000000000000000000000000dEaD, _tokenId, "");
   }
 
-  function sendToNewTitleEscrow(address beneficiary, address holder, uint256 _tokenId) public onlyMinter {
+  function sendToNewTitleEscrow(
+    address beneficiary,
+    address holder,
+    uint256 _tokenId
+  ) public onlyMinter {
     address newTitleEscrow = this.deployNewTitleEscrow(address(this), beneficiary, holder);
-    _safeTransferFrom(address(this), newTitleEscrow, _tokenId, "");
+    this.safeTransferFrom(address(this), newTitleEscrow, _tokenId, "");
```

Comment:
since the _safeTransferFrom (bypasses the check that if msg.sender is owner/approved) is removed, would this safeTransferFrom function prevent the transfer? since it is validating the msg.sender.